### PR TITLE
Added product type and shipping costs to snippet builder.

### PR DESCRIPTION
**Pull request by rtorres87**

### CODE REVIEW

File: src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java [hunk 1]
```diff
@@ -59,6 +61,10 @@
      * This flag is used to indicate that you want to show the failed payments of the current user.
      */
     private Boolean failedPayments;
+
+
+    private Double shippingCost;
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java]: - Double is not precise enough. We need BigDecimal here.
- Too many empty lines ;-)

File: src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java [hunk 2]
```diff
@@ -222,6 +229,24 @@ public CheckoutSnippetBuilder withCSRFToken(String token) {
...
+     * Adds the shipping cost to the snippet builder.
+     *
+     * @param shippingCost shipping cost of product
+     * @param productType  the product type related to the product
+     *
+     * @return The same {@link CheckoutSnippetBuilder} instance
+     *
+     * @throws PlenigoException if a validation error happens
+     */
+    public CheckoutSnippetBuilder withShipping(Double shippingCost, ProductType productType) throws PlenigoException {
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java]: BigDecimal

File: src/test/java/com/plenigo/sdk/builders/CheckoutSnippetBuilderTest.java [hunk 3]
```diff
@@ -218,13 +219,47 @@ public final void testFilledProduct() throws PlenigoException {
...
+        product.setCustomPrice(true);
+        CheckoutSnippetBuilder linkBuilder = getBuilder(product)
+                .withCSRFToken("TOKEN").withShipping(10.00, ProductType.BOOK);
+        String builtLink = linkBuilder.build();
+        assertNotNull(builtLink, "The generated link is null");
+        assertTrue("The link does not match the expected regex -> "
+                + builtLink, builtLink.matches(PLENIGO_CHECKOUT_BASE_REGEX));
+    }
+
+    private Product getProduct() {
```
Comment (maximilianschweitzer) [inline src/test/java/com/plenigo/sdk/builders/CheckoutSnippetBuilderTest.java]: Missing comment.

File: src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java (hunk 1 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java]: Fixed

File: src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java (hunk 2 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java]: Fixed

File: src/test/java/com/plenigo/sdk/builders/CheckoutSnippetBuilderTest.java (hunk 3 above)
Comment (rtorres87) [inline src/test/java/com/plenigo/sdk/builders/CheckoutSnippetBuilderTest.java]: Fixed

**Pull request merged**